### PR TITLE
Update small scale deployment

### PR DESCRIPTION
**Pull request by mmattel**
Fixes: [oCIS docker behind apache proxy](https://github.com/owncloud/ocis/issues/5214#top)

This PR:

* fixes and improves some things
* puts all examples into partial files
* adds a Apache section next to nginx

Language review welcomed.

Preview visible on staging.

@enbrnz fyi

Note that it is by purpose *not* using certbot command line options including --redirect.

### CODE REVIEW

File: modules/ROOT/partials/deployment/nginx_full_example.adoc [hunk 1]
```diff
@@ -28,7 +29,7 @@ server {
         ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;
 
         location / {
-            proxy_pass http://localhost:{ocis_port};
+            proxy_pass http://127.0.0.1:{ocis_port};
```
Comment (phil-davis) [inline modules/ROOT/partials/deployment/nginx_full_example.adoc:32]: hmmm - do we need to put a literal IP local-host address here?
What happens to IPv6 in this case? It's "localhost" equivalent address won't be covered?

File: modules/ROOT/pages/examples/small-scale.adoc [hunk 2]
```diff
@@ -229,6 +232,65 @@ Reload nginx again:
...
+
+Adapt the config to redirect automatically to https, use the newly generated certificates and add the required proxy configuration. The full config should look like this:
+
+include::partial$deployment/apache_full_example.adoc[]
+
+Add the following to the SSL definition above if SSL security should be hardened / improved:
+
+include::partial$deployment/apache_harden_ssl.adoc[]
+
+Reload nginx again:
```
Comment (EParzefall) [inline modules/ROOT/pages/examples/small-scale.adoc:285]:
```suggestion
Reload Apache:
```

[the remark this answers is not in the document]
Comment (EParzefall) [reply modules/ROOT/pages/examples/small-scale.adoc:285]: Right, I've fixed it.

File: modules/ROOT/partials/deployment/nginx_full_example.adoc (hunk 1 above)
Comment (mmattel) [reply modules/ROOT/partials/deployment/nginx_full_example.adoc:32]: ah, you are right, will change

Comment (EParzefall) [inline modules/ROOT/partials/deployment/nginx_full_example.adoc:32]:
```suggestion
            proxy_pass http://localhost:{ocis_port};
```

Comment (EParzefall) [reply modules/ROOT/partials/deployment/nginx_full_example.adoc:32]: Reverted

**Review by EParzefall — APPROVED** [review]

File: modules/ROOT/pages/examples/small-scale.adoc [hunk 3]
```diff
@@ -229,6 +232,65 @@ Reload nginx again:
...
+
+Add the following to the SSL definition above if SSL security should be hardened / improved:
+
+include::partial$deployment/apache_harden_ssl.adoc[]
+
+Reload Apache:
+
+[source,bash]
+----
+sudo systemctl reload nginx
```
Comment (phil-davis) [inline modules/ROOT/pages/examples/small-scale.adoc:289]: This looks wrong - it should be reloading/restarting Apache

**Review by phil-davis — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/examples/small-scale.adoc (hunk 3 above)
Comment (mmattel) [reply modules/ROOT/pages/examples/small-scale.adoc]: fixed

**Pull request merged**